### PR TITLE
Update dependencies and CI image

### PR DESCRIPTION
**Pull request by jsanz**
* Updates the CI buildkite agent to the last version
* Upgrades typescript, semver, husky, jest, eslint, lru-cache, prettier, maplibre, node-fetch
* Husky upgrade forced a new file with the script instead of the section in the `package.json`
* eslint upgrade needed a new configuration file, took the opportunity to simplify the configuration and take prettier totally out of the eslint scope.
* Most of code changes are coming from the TypeScript upgrade but also maplibre required some small fixes. I left comments on the relevant bits.

> [!note]
> Linked the package to ems-landing-page and kibana to test that the new deps where OK and (after downgrading lru-cache) everything works on my dev environments.

Replaces #231  and #230

### CODE REVIEW

File: src/ems_client.ts [hunk 1]
```diff
@@ -290,7 +290,11 @@ export class EMSClient {
       if (!response.ok) {
         throw new Error(response.statusText);
       }
-      return response ? await response.json() : null;
```
Comment (jsanz) [inline src/ems_client.ts:293]: change forced by the @types/node upgrade from `13.13` to `20.12`.

File: src/tms_service.ts [hunk 2]
```diff
@@ -374,33 +374,45 @@ export class TMSService extends AbstractEmsService {
 
   async getMinZoom(format = 'vector'): Promise<number | undefined> {
     switch (format) {
-      case 'vector':
+      case 'vector': {
         const { sources } = (await this._getVectorStyleJsonInlined()) || { sources: {} };
         return Math.min(
           ...Object.values(sources)
-            .map(({ minzoom }) => minzoom)
```
Comment (jsanz) [inline src/tms_service.ts:381]: code change from typescript upgrade, `minzoom` was not inferred from the parameter and needed to be explicitly checked.

File: src/tms_service.ts [hunk 3]
```diff
@@ -374,33 +374,45 @@ export class TMSService extends AbstractEmsService {
...
           ...Object.values(sources)
-            .map(({ maxzoom }) => maxzoom)
+            .map((s) => {
+              return s && s instanceof Object && 'maxzoom' in s ?
+                s['maxzoom'] :
+                null
+            })
             .filter((maxzoom): maxzoom is number => Number.isFinite(maxzoom))
         );
-      case 'raster':
```
Comment (jsanz) [inline src/tms_service.ts:401]: eslint upgrade complained about the code blocks missing the fences.

File: test/__snapshots__/ems_client.test.ts.snap [hunk 4]
```diff
@@ -1,37 +1,37 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`.getFileLayers[1] - getGeoJson defaults should convert topojson to geojson 1`] = `
-Object {
```
Comment (jsanz) [inline test/__snapshots__/ems_client.test.ts.snap:4]: `jest` required a refresh on the snapshots

File: tsconfig.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     /* Basic Options */
     // "incremental": true,                   /* Enable incremental compilation */
-    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */
```
Comment (jsanz) [inline tsconfig.json:5]: `lru-cache` code required to target at least to `es2015`. It seems OK to upgrade this given the library usage.

**Review by nickpeihl — APPROVED** [review]
lgtm! 

code review only.

**Pull request merged**